sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -439,6 +439,21 @@ export class EventHubSender extends LinkEntity {
     }
     const sendEventPromise = () =>
       new Promise<void>((resolve, reject) => {
+        const abortSignal: AbortSignalLike | undefined = options!.abortSignal;
```

Comment:
We can move this line out of the `sendEventPromise` right as it doesnt change during the retries? That way, we shouldnt need to use the `!` on the `options`